### PR TITLE
[ISSUE #3484] springmvc client lossless registration

### PR DESCRIPTION
**Pull request by gaochuang98**
Make sure that:

- [x] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientEventListener.java [hunk 1]
```diff
@@ -102,46 +103,42 @@ public SpringMvcClientBeanPostProcessor(final PropertiesConfig clientConfig,
...
+        for (Map.Entry<String, Object> entry : controllerBeans.entrySet()) {
+            handler(entry.getValue());
         }
-        
-        final ShenyuSpringMvcClient beanShenyuClient = AnnotatedElementUtils.findMergedAnnotation(clazz, ShenyuSpringMvcClient.class);
-        final String superPath = buildApiSuperPath(clazz);
+    }
+
+    private void handler(@NonNull final Object bean) {
+        final ShenyuSpringMvcClient beanShenyuClient = AnnotationUtils.findAnnotation(bean.getClass(), ShenyuSpringMvcClient.class);
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientEventListener.java:116]: 1. lost aop handle
```java
        Class<?> clazz = bean.getClass();
        if (AopUtils.isAopProxy(bean)) {
            clazz = AopUtils.getTargetClass(bean);
        }
```
2. should use `AnnotatedElementUtils#findMergedAnnotation`

File: shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientEventListener.java [hunk 2]
```diff
@@ -102,46 +103,42 @@ public SpringMvcClientBeanPostProcessor(final PropertiesConfig clientConfig,
...
             publisher.publishEvent(buildMetaDataDTO(beanShenyuClient, pathJoin(contextPath, superPath)));
-            return bean;
+            return;
         }
-        final Method[] methods = ReflectionUtils.getUniqueDeclaredMethods(clazz);
+        final Method[] methods = ReflectionUtils.getUniqueDeclaredMethods(bean.getClass());
         for (Method method : methods) {
             final RequestMapping requestMapping = AnnotatedElementUtils.findMergedAnnotation(method, RequestMapping.class);
-            ShenyuSpringMvcClient methodShenyuClient = AnnotatedElementUtils.findMergedAnnotation(method, ShenyuSpringMvcClient.class);
+            ShenyuSpringMvcClient methodShenyuClient = AnnotationUtils.findAnnotation(method, ShenyuSpringMvcClient.class);
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientEventListener.java:126]: should use `AnnotatedElementUtils#findMergedAnnotation`

File: shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientEventListener.java [hunk 3]
```diff
@@ -102,46 +103,42 @@ public SpringMvcClientBeanPostProcessor(final PropertiesConfig clientConfig,
...
-    
+
     private <A extends Annotation> boolean hasAnnotation(final @NonNull Class<?> clazz, final @NonNull Class<A> annotationType) {
         return Objects.nonNull(AnnotationUtils.findAnnotation(clazz, annotationType));
     }
-    
+
     private String buildApiPath(@NonNull final Method method, @NonNull final String superPath) {
-        ShenyuSpringMvcClient shenyuSpringMvcClient = AnnotatedElementUtils.findMergedAnnotation(method, ShenyuSpringMvcClient.class);
+        ShenyuSpringMvcClient shenyuSpringMvcClient = AnnotationUtils.findAnnotation(method, ShenyuSpringMvcClient.class);
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientEventListener.java:141]: should use `AnnotatedElementUtils#findMergedAnnotation`

File: shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientEventListener.java [hunk 4]
```diff
@@ -178,9 +175,9 @@ private String getPathByAnnotation(@Nullable final Annotation annotation, @NonNu
         }
         return null;
     }
-    
+
     private String buildApiSuperPath(@NonNull final Class<?> method) {
-        ShenyuSpringMvcClient shenyuSpringMvcClient = AnnotatedElementUtils.findMergedAnnotation(method, ShenyuSpringMvcClient.class);
+        ShenyuSpringMvcClient shenyuSpringMvcClient = AnnotationUtils.findAnnotation(method, ShenyuSpringMvcClient.class);
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientEventListener.java:180]: should use `AnnotatedElementUtils#findMergedAnnotation`

File: shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientEventListener.java (hunk 1 above)
Comment (gaochuang98) [reply shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientEventListener.java]: ok

File: shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientEventListener.java [hunk 5]
```diff
@@ -102,24 +104,26 @@ public SpringMvcClientBeanPostProcessor(final PropertiesConfig clientConfig,
         mappingAnnotation.add(RequestMapping.class);
         publisher.start(shenyuClientRegisterRepository);
     }
-    
+
     @Override
-    public Object postProcessAfterInitialization(@NonNull final Object bean, @NonNull final String beanName) throws BeansException {
+    public void onApplicationEvent(final ContextRefreshedEvent event) throws BeansException {
+        Map<String, Object> controllerBeans = event.getApplicationContext().getBeansWithAnnotation(Controller.class);
```
Comment (dragon-zhang) [inline shenyu-client/shenyu-client-http/shenyu-client-springmvc/src/main/java/org/apache/shenyu/client/springmvc/init/SpringMvcClientEventListener.java:110]: lost is full handle
```java
        if (Boolean.TRUE.equals(isFull)) {
            return bean;
        }
```

**Review by dragon-zhang — APPROVED** [review]

**Pull request merged**